block/manager.go

```diff
@@ -55,9 +55,10 @@ type Manager struct {
 	*/
 	// The last height which was submitted to both sublayers, that we know of. When we produce new batches, we will
 	// start at this height + 1.
-	// It is ALSO used by the producer, because the producer needs to check if it can prune blocks and it wont'
+	// It is ALSO used by the producer, because the producer needs to check if it can prune blocks and it won't
 	// prune anything that might be submitted in the future. Therefore, it must be atomic.
 	LastSubmittedHeight atomic.Uint64
+	TargetHeight        atomic.Uint64
```

Comment:
It's under the submission section but it's used in the retriever loop

Needs a docstring too, I don't understand how it relates (or doesn't) to targetSyncHeight